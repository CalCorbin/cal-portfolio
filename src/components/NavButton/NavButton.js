import React from 'react'
import './styles.css'
import PropTypes from 'prop-types'

NavButton.propTypes = {
  buttonText: PropTypes.string.isRequired,
}

export default function NavButton(props) {
  const { buttonText } = props

  return (
    <div>
      <div data-testid="nav-button" className="__button-container">
        <p>{buttonText}</p>
      </div>
    </div>
  )
}
