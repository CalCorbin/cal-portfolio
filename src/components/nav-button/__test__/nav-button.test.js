import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NavButton from '../index';

afterEach(cleanup);

it('should render the nav-button correctly', () => {
  const { getByTestId } = render(<NavButton buttonText="testme" />);
  expect(getByTestId('nav-button')).toHaveClass('__button-container');
});
