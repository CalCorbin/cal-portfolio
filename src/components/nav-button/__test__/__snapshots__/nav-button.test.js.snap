// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the nav-button snapshot 1`] = `
<div>
  <div
    className="__button-container"
    data-testid="nav-button"
  >
    <p />
  </div>
</div>
`;
